templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -771,156 +781,179 @@ contents:
-      #   slaves to re-activate, but it should be with our profiles since they
-      #   have higher priority
-      # - make sure that ovs-if-phys0 and its slaves, if any, are activated.
-      # - finally activate ovs-if-br-ex which holds the IP configuration.
-      connections=(br-ex ovs-if-phys0)
-      if [ -f "$extra_bridge_file" ]; then
-        connections+=(br-ex1 ovs-if-phys1)
-      fi
-      while IFS= read -r connection; do
-        if [[ $connection == *"$MANAGED_NM_CONN_SUFFIX" ]]; then
-          connections+=("$connection")
-        fi
-      done < <(nmcli -g NAME c)
-      connections+=(ovs-if-br-ex)
-      if [ -f "$extra_bridge_file" ]; then
-        connections+=(ovs-if-br-ex1)
+        # Remove bridges created by openshift-sdn
+        ovs-vsctl --timeout=30 --if-exists del-br br0
+
+        # Make sure everything is activated. Do it in a specific order:
+        # - activate br-ex first, due to autoconnect-slaves this will also
+        #   activate ovs-port-br-ex, ovs-port-phys0 and ovs-if-phys0. It is
+        #   important that ovs-if-phys0 activates with br-ex to avoid the
+        #   ovs-if-phys0 profile being overridden with a profile generated from
+        #   kargs. The activation of ovs-if-phys0, if a bond, might cause the
+        #   slaves to re-activate, but it should be with our profiles since they
+        #   have higher priority
+        # - make sure that ovs-if-phys0 and its slaves, if any, are activated.
+        # - finally activate ovs-if-br-ex which holds the IP configuration.
+        connections=(br-ex ovs-if-phys0)
+        if [ -f "$extra_bridge_file" ]; then
+          connections+=(br-ex1 ovs-if-phys1)
+        fi
+        while IFS= read -r connection; do
+          if [[ $connection == *"$MANAGED_NM_CONN_SUFFIX" ]]; then
+            connections+=("$connection")
+          fi
+        done < <(nmcli -g NAME c)
+        connections+=(ovs-if-br-ex)
+        if [ -f "$extra_bridge_file" ]; then
+          connections+=(ovs-if-br-ex1)
+        fi
+        activate_nm_connections "${connections[@]}"
+        try_to_bind_ipv6_address
+        set_nm_conn_files
+      elif [ "$1" == "OpenShiftSDN" ]; then
+        # Revert changes made by /usr/local/bin/configure-ovs.sh during SDN migration.
+        rollback_nm
+        
+        # Remove bridges created by ovn-kubernetes
+        ovs-vsctl --timeout=30 --if-exists del-br br-int -- --if-exists del-br br-local
       fi
-      activate_nm_connections "${connections[@]}"
-      try_to_bind_ipv6_address
-      set_nm_conn_files
-    elif [ "$1" == "OpenShiftSDN" ]; then
-      # Revert changes made by /usr/local/bin/configure-ovs.sh during SDN migration.
-      rollback_nm
-      
-      # Remove bridges created by ovn-kubernetes
-      ovs-vsctl --timeout=30 --if-exists del-br br-int -- --if-exists del-br br-local
-    fi
+    }
+
+    # Retry configure_ovs until it succeeds.
```

Comment:
WDYT about writing the number of times it retries to a file and then write a CI test case to detect multiple retries and fail?

Reply:
Rather that implementing a CI test for it, we should add the check to all ovn-k CI tests?
Because the failure itself would be something unexpected.
Would it be enough to touch a file when there has been a retry?

Reply:
+1

I think touch is perfectly fine.